include/envoy/config/incremental_subscription.h

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "envoy/common/exception.h"
+#include "envoy/common/pure.h"
+#include "envoy/stats/stats_macros.h"
+
+#include "common/protobuf/protobuf.h"
+
+namespace Envoy {
+namespace Config {
+
+template <class ResourceType> class IncrementalSubscriptionCallbacks {
+public:
+  typedef Protobuf::RepeatedPtrField<ResourceType> ResourceVector;
+
+  virtual ~IncrementalSubscriptionCallbacks() {}
+
+  /**
+   * Called when an incremental configuration update is received.
+   * @param added_resources resources newly added since the previous fetch.
+   * @param removed_resources names of resources that this fetch instructed to be removed.
+   * @param version_info update version.
```

Comment:
This is presumably the system version, vs. the resource version, since that is inside the resource vectors. It wouldn't hurt to be more specific here in naming to minimize the confusion whenever using version, i.e. `system_version` vs. `resource_version`.

Reply:
Done.